tracer/src/Datadog.Trace/Debugger/SpanCodeOrigin/EndpointDetector.cs

```diff
@@ -0,0 +1,296 @@
+// <copyright file="EndpointDetector.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Datadog.Trace.Pdb;
+using Datadog.Trace.VendoredMicrosoftCode.System.Collections.Immutable;
+using Datadog.Trace.VendoredMicrosoftCode.System.Reflection.Metadata;
+using Datadog.Trace.VendoredMicrosoftCode.System.Reflection.Metadata.Ecma335;
+
+namespace Datadog.Trace.Debugger.SpanCodeOrigin;
+
+internal static class EndpointDetector
+{
+    private static readonly HashSet<string> ControllerAttributes =
+    [
+        "Microsoft.AspNetCore.Mvc.ApiControllerAttribute",
+        "Microsoft.AspNetCore.Mvc.ControllerAttribute",
+        "Microsoft.AspNetCore.Mvc.RouteAttribute"
+    ];
+
+    private static readonly HashSet<string> ControllerBaseNames =
+    [
+        "Microsoft.AspNetCore.Mvc.Controller",
+        "Microsoft.AspNetCore.Mvc.ControllerBase"
+    ];
+
+    private static readonly HashSet<string> ActionAttributes =
+    [
+        "Microsoft.AspNetCore.Mvc.HttpGetAttribute",
+        "Microsoft.AspNetCore.Mvc.HttpPostAttribute",
+        "Microsoft.AspNetCore.Mvc.HttpPutAttribute",
+        "Microsoft.AspNetCore.Mvc.HttpDeleteAttribute",
+        "Microsoft.AspNetCore.Mvc.HttpPatchAttribute",
+        "Microsoft.AspNetCore.Mvc.HttpHeadAttribute",
+        "Microsoft.AspNetCore.Mvc.HttpOptionsAttribute",
+    ];
+
+    private static readonly HashSet<string> SignalRHubBaseNames =
+    [
+        "Microsoft.AspNetCore.SignalR.Hub",
+        "Microsoft.AspNetCore.SignalR.Hub`1"
+    ];
+
+    private static readonly HashSet<string> PageModelBaseNames = ["Microsoft.AspNetCore.Mvc.RazorPages.PageModel"];
+
+    private static readonly HashSet<string> NoHandlerAttributes = ["Microsoft.AspNetCore.Mvc.RazorPages.NonHandlerAttribute"];
+
+    public static ImmutableHashSet<int> GetEndpointMethodTokens(DatadogMetadataReader datadogMetadataReader)
+    {
+        var builder = ImmutableHashSet.CreateBuilder<int>();
+        var metadataReader = datadogMetadataReader.MetadataReader;
+        foreach (var typeHandle in metadataReader.TypeDefinitions)
+        {
+            var typeDef = metadataReader.GetTypeDefinition(typeHandle);
+
+            if (!IsValidTypeKind(typeDef))
+            {
+                continue;
+            }
+
+            bool isPageModel = false, isSignalRHub = false, isCompilerGeneratedType = false;
```

Comment:
nit: 100% just style preferences, but we don't tend to use this style in dd-dotnet, so better to be consistent I think.
```suggestion
            bool isPageModel = false;
            bool isSignalRHub = false;
            bool isCompilerGeneratedType = false;
```

Reply:
Fixed in a5a2c085c57f23a246a2b781dec1bc9354502a16